src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -574,12 +577,20 @@ private String generateReplyAddress() {
       }
       if (subs != null && !replyHandler && !localOnly) {
         // Propagate the information
-        subs.add(address, serverID, registration::setResult);
+        subs.add(address, serverID, ar -> {
+          holder.context.runOnContext(v -> {
+            registration.setResult(ar);
+          });
+        });
       } else {
-        registration.setResult(Future.succeededFuture());
+        holder.context.runOnContext(v -> {
```

Comment:
actually it needs to be done when the context has just been created (embedded) example.